.github/workflows/flow-pull-request-checks.yaml

```diff
@@ -0,0 +1,39 @@
+name: PR Checks
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+concurrency:
+  group: pr-checks-${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  test:
+    name: Tests
+    runs-on: [self-hosted, Linux, medium, ephemeral]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

Comment:
We should be binding these to the SHA commit hash and not the floating version tags.